tests/Platforms/PostgreSQLPlatformTest.php

```diff
@@ -304,14 +305,22 @@ protected function getQuotedColumnInForeignKeySQL(): array
         ];
     }
 
-    #[DataProvider('pgBooleanProvider')]
+    #[DataProvider('provideConvertBooleansAsLiteralStrings')]
     public function testConvertBooleanAsLiteralStrings(
         string $databaseValue,
         string $preparedStatementValue,
     ): void {
         self::assertEquals($preparedStatementValue, $this->platform->convertBooleans($databaseValue));
     }
 
+    /** @return Generator<int, array{string, string}> */
+    public static function provideConvertBooleansAsLiteralStrings(): Generator
+    {
+        foreach (self::pgBooleanProvider() as $key => $params) {
+            yield $key => [$params[0], $params[1]];
+        }
+    }
```

Comment:
PHPUnit 12 complains if the data provider provides more arguments than the test consumes. This change could be backported to 4.x to make the diff smaller.